kornia/geometry/warp/homography_warper.py

```diff
@@ -295,6 +295,9 @@ def normal_transform_pixel(height: int, width: int) -> torch.Tensor:
     tr_mat[0, 0] = tr_mat[0, 0] * 2.0 / (width - 1.0)
     tr_mat[1, 1] = tr_mat[1, 1] * 2.0 / (height - 1.0)
```

Comment:
Is it better to have an epsilon here?

Reply:
I tried adding epsilon first, but then it broke a bunch of normal transformation tests-- because it adds the epsilon in all cases. That's why I changed it to a check for `isinf`, so that it only changes values in that case.

Reply:
How about something like ```eps if height == 1. else hight - 1.```

Reply:
Sure, but then you'd have to do the if / else for all inputs, rather than the one indexing line. We can change the float there to some `eps` variable of course.

Reply:
> Sure, but then you'd have to do the if / else for all inputs, rather than the one indexing line.

I did not get it. I think it shall be something like :
```python
height = 1e-14 if height == 1. else hight - 1
width = 1e-14 if width == 1. else width - 1
```
Then we replace the following lines as:
```python
tr_mat[0, 0] = tr_mat[0, 0] * 2.0 / width
tr_mat[1, 1] = tr_mat[1, 1] * 2.0 / height
```

Reply:
@dwiel have you played with different epsilon values? How bad were the tests?

Reply:
With epsilon `1.19e-7`, about 30 tests failed. However, if we only add it in the case of `dimension==1` like @shijianjian suggested, the tests would've passed

Reply:
I see. Ugly hack, but if it works fair enough. I would try to add small unit tests for this function that reflects this issue.